### PR TITLE
DCMAW-8008: Add Splash Screen

### PR DESCRIPTION
**Pull request by alex-bradbury**
- Add splash screen composable
- Add splash screen routing to `LoginRoutes`
- `SplashScreenViewModel` to handle logging in and show/not show unlock button
- Move Login logic out of `MainActivityViewModel` into `HandleLogin` usecase
- Usecase is now used in splash view model and main activity view model

### Handling the SecureStore now using Flow

- Convert the `SecureStoreErrorType` to a `LocalAuthStatus` in `GetFromSecureStore` use case
- Use LocalAuthStatus to handle routing/errors rather than thrown exceptions
- Introduce callbacks for `GetFromSecureStore` and `HandleLogin`
- Currently do nothing with SecureStoreError and BioCheckFailed in the view model
- Remove StateGenerator (not sure it was doing anything....)

Resolves: DCMAW-8008

### CODE REVIEW

File: app/src/main/java/uk/gov/onelogin/MainActivity.kt [hunk 1]
```diff
@@ -44,10 +37,9 @@ class MainActivity : AppCompatActivity() {
                     next.observe(lifecycleOwner) {
                         navController.navigate(it)
                     }
-                    handleIntent(
-                        intent = intent,
-                        fragmentActivity = this@MainActivity
-                    )
+                    if (intent?.data != null) {
```
[the remark this answers is not in the document]
Comment (alex-bradbury) [reply app/src/main/java/uk/gov/onelogin/MainActivity.kt:40]: yep, removed

File: app/src/androidTest/java/uk/gov/onelogin/ui/splash/SplashScreenViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            viewModel.login(composeTestRule.activity as FragmentActivity)
+
+            Handler(Looper.getMainLooper()).post {
+                assertEquals(HomeRoutes.START, viewModel.next.value)
+            }
+        }
+    }
+
+    @Test
+    fun loginThrowsGeneral() {
```
[the remark this answers is not in the document]
Comment (alex-bradbury) [reply app/src/androidTest/java/uk/gov/onelogin/ui/splash/SplashScreenViewModelTest.kt:72]: mmmm yeah good point, perhaps we should send them to the login screen

[the remark this answers is not in the document]
Comment (flavius-mester) [reply app/src/androidTest/java/uk/gov/onelogin/ui/splash/SplashScreenViewModelTest.kt]: I have tested the changes, and actually results in a new bug :)) 
It results in the user still seeing the fingerprint scanner (retry) but the screen underneath is Login.

So, I think we need to distinguish the RetryErrors from Generic ones 
9AA0E81B-19AE-4462-8B5D-71887F46A210_1_105_c

Maybe reverting this back will be better for now and perhaps addressing this in a different ticket?

**Comment by flavius-mester** [comment]
The SplashScreen Theme is not full screen anymore?

IMG_2505

File: app/src/androidTest/java/uk/gov/onelogin/ui/splash/SplashScreenViewModelTest.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (alex-bradbury) [reply app/src/androidTest/java/uk/gov/onelogin/ui/splash/SplashScreenViewModelTest.kt]: Yep, reverted!

**Review by flavius-mester — APPROVED** [review]

**Pull request merged**